### PR TITLE
JAMES-3157 Add support for bench profiles

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: benchmarks/s3.properties [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# ============================================= BlobStore Implementation ==================================
```
Comment (chibenwa) [inline benchmarks/s3.properties:1]: May I ask where this file is used?

(I did not understood)

File: benchmarks/s3.properties [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+# Choose your BlobStore implementation
+# Mandatory, allowed values are: cassandra, objectstorage, hybrid
+# hybrid is using both objectstorage for unfrequently read or big blobs & cassandra for small, often read blobs
+implementation=objectstorage
+
+# ========================================= Hybrid BlobStore ======================================
+# hybrid is using both objectstorage for unfrequently read or big blobs & cassandra for small, often read blobs
+# Size threshold for considering a blob as 'big', causing it to be saved in the low cost blobStore
+# Optional, defaults to 32768 bytes (32KB), must be positive
+hybrid.size.threshold=32768
```
Comment (chibenwa) [inline benchmarks/s3.properties:13]: Likely not needed.

File: benchmarks/swift.properties [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+# Choose your BlobStore implementation
+# Mandatory, allowed values are: cassandra, objectstorage, hybrid
+# hybrid is using both objectstorage for unfrequently read or big blobs & cassandra for small, often read blobs
+implementation=objectstorage
+
+# ========================================= Hybrid BlobStore ======================================
+# hybrid is using both objectstorage for unfrequently read or big blobs & cassandra for small, often read blobs
+# Size threshold for considering a blob as 'big', causing it to be saved in the low cost blobStore
+# Optional, defaults to 32768 bytes (32KB), must be positive
+hybrid.size.threshold=0
```
Comment (chibenwa) [inline benchmarks/swift.properties:13]: Likely not needed.

File: benchmarks/s3.properties (hunk 1 above)
Comment (blackheaven) [reply benchmarks/s3.properties:1]: it may be called in the groovy

Comment (chibenwa) [reply benchmarks/s3.properties:1]: `-v benchmarks/${params.PROFILE}.properties:/root/conf/blobs.properties`

By the monts, got it.

**Comment by Arsnael** [comment]
I still don't get where PROFILE variable is explicit enough that it's about blobstores

**Comment by blackheaven** [comment]
> I still don't get where PROFILE variable is explicit enough that it's about blobstores

because that's not only about blob store, in case of 'reference', we reload a dataset, while we do not otherwise.

**Comment by Arsnael** [comment]
> because that's not only about blob store, in case of 'reference', we reload a dataset, while we do not otherwise.

I still don't see the relation between profile and data, but I don't have better to propose so ok

**Comment by blackheaven** [comment]
:red_circle: on a non-built file...

test this please

**Comment by chibenwa** [comment]
test this please

**Comment by mbaechler** [comment]
> > because that's not only about blob store, in case of 'reference', we reload a dataset, while we do not otherwise.
> 
> I still don't see the relation between profile and data, but I don't have better to propose so ok

See my proposal there https://github.com/linagora/james-jenkins2/pull/13#discussion_r416489873

**Comment by Arsnael** [comment]
```
[afa1ee91d1f9f152a9c1a80eade7050771c62974] [ERROR] Errors: 
[afa1ee91d1f9f152a9c1a80eade7050771c62974] [ERROR]   MemorySpamAssassinContractTest.expungingSpamMessageShouldNotImpactSpamAssassinState » ConditionTimeout
```
test this please

File: JenkinsfileStressTests.groovy [hunk 4]
```diff
@@ -55,21 +60,42 @@ pipeline {
...
                                     sleep 5
-                                    def jamesCliWithOptions = 'java -jar /root/james-cli.jar -h 127.0.0.1 -p 9999'
                                     sh "docker exec james_run ${jamesCliWithOptions} listusers"
                                 }
                             }
+                            if (params.PROFILE in ["s3", "swift"]) {
+                                sh "docker exec james_run ${jamesCliWithOptions} removedomain localhost"
+                                sh "docker exec james_run ${jamesCliWithOptions} removedomain james.linagora.com"
+                                sh "docker exec james_run ${jamesCliWithOptions} adddomain open-paas.org"
+                                for (int n = 0; n <= 100; n++) {
```
Comment (rouazana) [inline JenkinsfileStressTests.groovy:95]: do you think we could get these information (`open-paas.org` and `100`) from the gatling scenario? Or maybe it's worth to setup a common env variable?

**Pull request merged**